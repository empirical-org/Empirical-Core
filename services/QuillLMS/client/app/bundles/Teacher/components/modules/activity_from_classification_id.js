export default (classificationId) => {
  const intClassificationId = parseInt(classificationId);
  if (intClassificationId === 1) {
    return 'flag';
  } else if (intClassificationId === 2) {
    return 'puzzle';
  } else if (intClassificationId === 4) {
    return 'diagnostic';
  } else if (intClassificationId === 5) {
    return 'connect';
  } else if (intClassificationId === 6) {
    return 'lessons';
  } else if (intClassificationId === 9) {
    return 'comprehension';
  }
};
