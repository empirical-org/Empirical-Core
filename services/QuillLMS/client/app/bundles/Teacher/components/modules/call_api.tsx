import request from 'request';
import getAuthToken from './get_auth_token';

function put(path: string, payload: object, onSuccess: Function, onError: Function): void {
  let uri = `${window.location.protocol}//${window.location.host}${path}`;
  let authedPayload = Object.assign(payload, {authenticity_token: getAuthToken()});
  request({
    method: 'PUT',
    uri: uri,
    json: authedPayload,
  }, (error, response) => {
    if (error) {
      // TODO: Replace this with some sort of real error reporting
      throw error;
    }
    switch (response.statusCode) {
      case 200:
        if (onSuccess) onSuccess(response);
        break;
      case 422:
        if (onError) onError(response);
        break;
    }
  });
}

export function changeActivityPackName(activityPackId: Number, name: string, onSuccess: Function, onError: Function): void {
  let uri = `/teachers/units/${activityPackId}`;
  let payload = {unit: {name: name}};
  put(uri, payload, onSuccess, onError);
}

export function changeActivityPackOrder(activityPackId: Number, activityIds: Array<Number>, onSuccess: Function, onError: Function): void {
  let uri = `/teachers/units/${activityPackId}/update_activities`;
  let payload = {data: {activities_data: activityIds.map((id) => { return {id: id} })}};
  put(uri, payload, onSuccess, onError);
}
