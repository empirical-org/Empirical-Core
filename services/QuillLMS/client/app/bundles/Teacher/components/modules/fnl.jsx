'use strict'

import _ from 'underscore'

export default  function () {

  this.toggle = function (array, item) {
    var newArray, alreadyThere;
    alreadyThere = _.contains(array, item)
    if (alreadyThere) {
      newArray = _.reject(array, function (ele) {return ele == item});
    } else {
      newArray = _.chain(array).push(item).value();
    }
    return newArray;
  };

  this.toggleById = function (array, item) {
    var newArray, alreadyThere, extant;
    extant = _.find(array, function (ele) { return ele.id === item.id })
    alreadyThere = (extant != undefined)
    if (alreadyThere) {
      newArray = _.reject(array, function (ele) {return ele == item})
    } else {
      newArray = _.chain(array).push(item).value()
    }
    return newArray;
  }
}
