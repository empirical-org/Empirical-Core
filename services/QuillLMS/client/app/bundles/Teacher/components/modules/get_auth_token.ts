export default () => {
  const csrfToken: HTMLElement | null =
    document.getElementsByName("csrf-token")[0];

  return csrfToken instanceof HTMLMetaElement ? csrfToken.content : 0;
};
