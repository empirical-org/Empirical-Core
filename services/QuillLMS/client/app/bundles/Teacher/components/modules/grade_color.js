export default (grade) => {
  if (grade == null) {
    return 'gray';
  } else if (grade < 0.32) {
    return 'red';
  } else if (grade < 0.83) {
    return 'orange';
  } else if (grade <= 1.0) {
    return 'green';
  }
  return 'gray';
};
