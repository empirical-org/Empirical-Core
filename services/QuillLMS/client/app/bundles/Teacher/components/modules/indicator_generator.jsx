'use strict';
export default function (getModelState, setModelState, fnl) {

  this.stateItemToggler = function (key) {
    return function (item, boolean) {
      let x = fnl.toggle(getModelState(key), item);
      setModelState(key, x);
    }
  };

  this.selector = function (key) {
    return function (value) {
      setModelState(key, value);
    }
  };
}
