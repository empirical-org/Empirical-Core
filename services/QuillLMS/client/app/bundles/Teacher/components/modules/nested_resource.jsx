import _ from 'underscore'

export default function (component, kind) {
  // returns a set of data, options
  this.add = function (resource) {
    let newModel = component.state.model
    let newNest = _.chain(newModel[kind]).push(resource).value()
    newModel[kind] = newNest;
    return newModel
  }


  this.remove = function (resource) {
    let newModel = component.state.model
    let newNest = _.reject(newModel[kind], function (r) { return r.id == resource.id })
    newModel[kind] = newNest;
    return newModel
  }
}
