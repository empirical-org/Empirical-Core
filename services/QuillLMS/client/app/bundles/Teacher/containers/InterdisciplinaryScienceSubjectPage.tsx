import * as React from 'react';

import QuestionsAndAnswers from './QuestionsAndAnswers'

import SubjectPageTeacherCenterArticles from '../components/content_hubs/subject_page_teacher_center_articles'
import SubjectPageSummary from '../components/content_hubs/subject_page_summary'
import { LARGE_ICON_BASE_SRC, } from '../../Shared/index'

const aiIconSrc = `${LARGE_ICON_BASE_SRC}/ai.svg`

const pageImgBaseSrc = `${process.env.CDN_URL}/images/pages/interdisciplinary_science_subject_page`

const quillPlusAiEduLogo = `${pageImgBaseSrc}/quill_logo_plus_ai_edu_logo.svg`

const BUILDING_AI_KNOWLEDGE = `${window.location.pathname}/building-ai-knowledge`.replace('//', '/')

const overview = (
  <header>
    <div className="container">
      <h1>Interdisciplinary Science</h1>
      <p>Over the next few years, Quill will building out Reading for Evidence activities aligned to high school science classrooms. Our first offering, Building AI Knowledge, features activities that introduce students to how AI will impact society over the next decade. In the future, we will build offerings around biology, earth sciences, and other science topics.<br /><br />In these Reading for Evidence activities, students read science-aligned texts, use what they’ve read to draft evidence-based claims, and receive AI-powered feedback to strengthen their writing. Process and growth are key: our responsive coaching guides students to revise until their sentences are clear, logical, and grounded in specific details from the text.<br /><br />Our Interdisciplinary Science work is meant to be used in either ELA or science classrooms. These activities are not meant to replace a textbook. Our activities provide a deep-dive into some of the most fascinating ideas in science.</p>
      <div className="course-cards">
        <div className="course-card">
          <img alt="" src={aiIconSrc} />
          <div>
            <h2>Building AI Knowledge</h2>
          </div>
          <a className="quill-button focus-on-light contained medium teal" href={BUILDING_AI_KNOWLEDGE}>View activities</a>
        </div>
      </div>
    </div>
  </header>
)

const summary = (
  <SubjectPageSummary
    paragraphCopy="Written and reviewed by educators, each Quill Interdisciplinary Science activity highlights a science concept. Not only do these activities deepen students’ subject area learning and reinforce essential science content, they also provide support for:"
  />
)

const partners = (
  <section className="partners container">
    <img alt="Quill logo plus AI Edu logo" src={quillPlusAiEduLogo} />
    <p>Quill has partnered with <a href="https://www.aiedu.org/" rel="noopener noreferrer" target="_blank">aiEDU</a> to curate activities from their site that work well with Quill Building AI Knowledge activities. On the course landing page, each Reading for Evidence activity includes a suggested paired aiEDU activity, which range from 5-minute warm-up activities to 3-4 hour student-led activities. These activities can be used to preview new content, reinforce prior knowledge, provide additional context, or, when used in conjunction with the Quill activity, serve as part of a text set to support extended writing practice. Whether used on their own or in conjunction with aiEDU activities, these resources are a valuable addition to any high school ELA or Computer Science classroom.</p>
  </section>
)

const tryItOutForYourself = (
  <section className="try-it-out">
    <div className="container">
      <h2>Try it out for yourself</h2>
      <div className="links-wrapper">
        <a className="quill-button focus-on-dark outlined teal large" href="/evidence/#/play?anonymous=true&uid=442" rel="noopener noreferrer" target="_blank" >Try a sample activity</a>
        <a className="quill-button focus-on-dark outlined teal large" href={BUILDING_AI_KNOWLEDGE}>View Building AI Knowledge activities</a>
      </div>
    </div>
  </section>
)

const InterdisciplinaryScienceSubjectPage = ({}) => {
  return (
    <div className="interdisciplinary-science-subject-page subject-page">
      {overview}
      {summary}
      {partners}
      <QuestionsAndAnswers questionsAndAnswersFile="interdisciplinaryScience" supportLink={null} />
      <SubjectPageTeacherCenterArticles />
      {tryItOutForYourself}
    </div>
  )
}

export default InterdisciplinaryScienceSubjectPage
