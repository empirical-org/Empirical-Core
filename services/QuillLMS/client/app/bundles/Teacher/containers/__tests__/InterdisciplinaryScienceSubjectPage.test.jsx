import * as React from 'react';
import { render } from '@testing-library/react';

import InterdisciplinaryScienceSubjectPage from '../InterdisciplinaryScienceSubjectPage';
import * as requestsApi from '../../../../modules/request';

describe('InterdisciplinaryScienceSubjectPage', () => {
  test('renders correctly', async () => {
    const { asFragment } = render(<InterdisciplinaryScienceSubjectPage />);

    expect(asFragment()).toMatchSnapshot();
  })
})
