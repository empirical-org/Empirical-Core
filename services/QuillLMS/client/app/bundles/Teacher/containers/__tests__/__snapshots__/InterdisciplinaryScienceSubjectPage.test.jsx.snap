// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterdisciplinaryScienceSubjectPage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="interdisciplinary-science-subject-page subject-page"
  >
    <header>
      <div
        class="container"
      >
        <h1>
          Interdisciplinary Science
        </h1>
        <p>
          Over the next few years, Quill will building out Reading for Evidence activities aligned to high school science classrooms. Our first offering, Building AI Knowledge, features activities that introduce students to how AI will impact society over the next decade. In the future, we will build offerings around biology, earth sciences, and other science topics.
          <br />
          <br />
          In these Reading for Evidence activities, students read science-aligned texts, use what they’ve read to draft evidence-based claims, and receive AI-powered feedback to strengthen their writing. Process and growth are key: our responsive coaching guides students to revise until their sentences are clear, logical, and grounded in specific details from the text.
          <br />
          <br />
          Our Interdisciplinary Science work is meant to be used in either ELA or science classrooms. These activities are not meant to replace a textbook. Our activities provide a deep-dive into some of the most fascinating ideas in science.
        </p>
        <div
          class="course-cards"
        >
          <div
            class="course-card"
          >
            <img
              alt=""
              src="undefined/images/icons/l/ai.svg"
            />
            <div>
              <h2>
                Building AI Knowledge
              </h2>
            </div>
            <a
              class="quill-button focus-on-light contained medium teal"
              href="/building-ai-knowledge"
            >
              View activities
            </a>
          </div>
        </div>
      </div>
    </header>
    <section
      class="summary-section"
    >
      <div
        class="container"
      >
        <h2>
          Build Content Knowledge & Writing Skills With Reading for Evidence
        </h2>
        <p>
          Written and reviewed by educators, each Quill Interdisciplinary Science activity highlights a science concept. Not only do these activities deepen students’ subject area learning and reinforce essential science content, they also provide support for:
        </p>
        <div
          class="summary-items"
        >
          <div
            class="summary-item"
          >
            <img
              alt=""
              src="undefined/images/icons/xl/search-document.svg"
            />
            <h3>
              Close reading
            </h3>
            <p>
              Every activity begins with a structured highlighting task as an entry point to the text. Then, students read and reread up to ten times in the process of developing their responses.
            </p>
          </div>
          <div
            class="summary-item"
          >
            <img
              alt=""
              src="undefined/images/icons/xl/research-books.svg"
            />
            <h3>
              Constructing evidence-based claims
            </h3>
            <p>
              Students receive custom feedback developed by real teachers to help them use what they have read to craft strong sentences.
            </p>
          </div>
          <div
            class="summary-item"
          >
            <img
              alt=""
              src="undefined/images/icons/xl/comment-document.svg"
            />
            <h3>
              Sentence fundamentals
            </h3>
            <p>
              Practice with conjunctions like because, but, and so provides students with a framework to combine sentences and express complex ideas effectively. Once they’ve clearly captured these ideas in their response, students receive grammar and spelling guidance too.
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="partners container"
    >
      <img
        alt="Quill logo plus AI Edu logo"
        src="undefined/images/pages/interdisciplinary_science_subject_page/quill_logo_plus_ai_edu_logo.svg"
      />
      <p>
        Quill has partnered with
        <a
          href="https://www.aiedu.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          aiEDU
        </a>
         to curate activities from their site that work well with Quill Building AI Knowledge activities. On the course landing page, each Reading for Evidence activity includes a suggested paired aiEDU activity, which range from 5-minute warm-up activities to 3-4 hour student-led activities. These activities can be used to preview new content, reinforce prior knowledge, provide additional context, or, when used in conjunction with the Quill activity, serve as part of a text set to support extended writing practice. Whether used on their own or in conjunction with aiEDU activities, these resources are a valuable addition to any high school ELA or Computer Science classroom.
      </p>
    </section>
    <div
      id="q-and-a"
    >
      <div
        class="q-and-a-inner-wrapper"
      >
        <h2>
          Questions and Answers
        </h2>
        <div
          class="qa-section"
        >
          <div
            class="qa"
          >
            <button
              class="question"
              tabindex="-1"
              type="button"
            >
              How does Quill Reading for Evidence build reading, writing, and critical thinking skills?
            </button>
          </div>
          <div>
            <button
              class="expand-collapse-button focus-on-light"
              type="button"
            >
              <p>
                Expand
              </p>
            </button>
          </div>
        </div>
        <div
          class="qa-section"
        >
          <div
            class="qa"
          >
            <button
              class="question"
              tabindex="-1"
              type="button"
            >
              What distinguishes Quill Reading for Evidence from other online reading tools?
            </button>
          </div>
          <div>
            <button
              class="expand-collapse-button focus-on-light"
              type="button"
            >
              <p>
                Expand
              </p>
            </button>
          </div>
        </div>
        <div
          class="qa-section"
        >
          <div
            class="qa"
          >
            <button
              class="question"
              tabindex="-1"
              type="button"
            >
              What do students need to know about Quill Reading for Evidence?
            </button>
          </div>
          <div>
            <button
              class="expand-collapse-button focus-on-light"
              type="button"
            >
              <p>
                Expand
              </p>
            </button>
          </div>
        </div>
        <div
          class="qa-section"
        >
          <div
            class="qa"
          >
            <button
              class="question"
              tabindex="-1"
              type="button"
            >
              What are some best practices for introducing Quill Reading for Evidence to my students and integrating it into my instruction?
            </button>
          </div>
          <div>
            <button
              class="expand-collapse-button focus-on-light"
              type="button"
            >
              <p>
                Expand
              </p>
            </button>
          </div>
        </div>
        <div
          class="qa-section"
        >
          <div
            class="qa"
          >
            <button
              class="question"
              tabindex="-1"
              type="button"
            >
              For whom are these activities most appropriate? How does the tool support students who may need more scaffolding, like multilingual learners and students with IEPs?
            </button>
          </div>
          <div>
            <button
              class="expand-collapse-button focus-on-light"
              type="button"
            >
              <p>
                Expand
              </p>
            </button>
          </div>
        </div>
        <a
          class="support-link hidden"
        >
          View all questions and answers
        </a>
      </div>
    </div>
    <section
      class="teacher-center-articles"
    >
      <div
        class="container"
      >
        <h2>
          Learn more with these Teacher Center articles
        </h2>
        <div
          class="articles-wrapper"
        >
          <span />
          <span />
        </div>
      </div>
    </section>
    <section
      class="try-it-out"
    >
      <div
        class="container"
      >
        <h2>
          Try it out for yourself
        </h2>
        <div
          class="links-wrapper"
        >
          <a
            class="quill-button focus-on-dark outlined teal large"
            href="/evidence/#/play?anonymous=true&uid=442"
            rel="noopener noreferrer"
            target="_blank"
          >
            Try a sample activity
          </a>
          <a
            class="quill-button focus-on-dark outlined teal large"
            href="/building-ai-knowledge"
          >
            View Building AI Knowledge activities
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
